powerline/segments/common/time.py

```diff
@@ -1,26 +1,38 @@
 # vim:fileencoding=utf-8:noet
 from __future__ import (unicode_literals, division, absolute_import, print_function)
 
+import os
 from datetime import datetime
 
 
-def date(pl, format='%Y-%m-%d', istime=False):
+def date(pl, format='%Y-%m-%d', istime=False, timezone=None):
 	'''Return the current date.
 
 	:param str format:
 		strftime-style date format string
 	:param bool istime:
 		If true then segment uses ``time`` highlight group.
+	:param str timezone:
+		A Timezone string. A list of timezones can usually be
+		obtained by listing the contents of
+		``/usr/share/zoneinfo/``. For example: ``Europe/Lisbon``.
 
 	Divider highlight group used: ``time:divider``.
 
 	Highlight groups used: ``time`` or ``date``.
 	'''
-	return [{
-		'contents': datetime.now().strftime(format),
-		'highlight_groups': (['time'] if istime else []) + ['date'],
-		'divider_highlight_group': 'time:divider' if istime else None,
-	}]
+	if timezone is not None:
+		old_tz = os.environ.get('TZ', None)
+		os.environ['TZ'] = timezone
+	try:
+		return [{
+			'contents': datetime.now().strftime(format),
+			'highlight_groups': (['time'] if istime else []) + ['date'],
+			'divider_highlight_group': 'time:divider' if istime else None,
+		}]
+	finally:
+		if timezone is not None and old_tz is not None:
+			os.environ['TZ'] = old_tz
```

Comment:
@s0undt3ch It is absolutely useless. It is possible that change to the environment will be reset *before* datetime runs because it will be reset *in another thread*. You cannot use locks here because `os.environ` is used in far too many places, mostly implicitly. I do not want powerline behaviour to be unstable.